### ISSUE TITLE
Guided Missile Launcher.

### ISSUE THREAD

**Issue by Gregory-Y** [deferred-impact/battledudes-issue-tracker#117]
1 magazine capacity (as expected from a launcher) with 2 reserve ammo (idk might change). About the same damage and aoe as the unguided rocket, smaller projectile. The missile is manually-guided, meaning the player controls it, the player's camera immediately shifts from the player to the missile after it is fired.

I have two ideas for actually controlling the missile, the first idea is as follows: and the missile has to follow the cursor. The player can still move but they cannot do pretty much anything else like switching weapons or using specials.

Alternatively, the player could be made completely stationary but the missile can be controlled by wasd or a joystick.

The missile travels at around 45, slightly slower than the flintlock bullet, and is about as maneuverable as a jeep. Also, you can only control the missile for a maximum of ~ 6-10 seconds before it automatically explodes due to running out of fuel (idk). If the player gets killed the missile just continues going in a straight line in whatever direction it was going previously, exploding after some time. Additionally, the missile operator can automatically disengage from the missile if they feel that they are in danger by pressing x, which also causes the missile to go in whatever direction its facing.

The missile can hit any obstacle and can be intercepted by the trophy system.

**Comment by Haox09** [comment]
This would be difficult as this is the first time in bd the center of screen wont be the player and be the rocket if i understood correctly. Alternative would be missil can only be controled when its on screen of the player. after that it will still be controled(or not?) but the player wont see it on their screen.

**Comment by Gregory-Y** [comment]
If Sarunas can't code the game so that the camera shifts to the rockets position then yeah we can do that.

**Comment by Profugus1** [comment]
The missile should explode after some time after firing and should be able to be shot down by other players

**Comment by Gregory-Y** [comment]
If you make the missile hittable by players' bullets, you'd want to make it smaller so that it doesn't get shot down too easily and doesn't act as a bullet shield. Also how much health do you think the missile should have? Oh and funny idea the missile's speed should be adjustable.

**Comment by deferred-impact** [comment]
> the player's camera immediately shifts from the player to the missile after it is fired.

I rather hate this. If other people prefer this, let them have it, but i'd like my camera to always be focused on my player; i'll gladly take the effective range L for it

(Plus, i don't want to be sniped from halfway across the map by such a missile, lmao)

Always having the camera focused on the player also removes the need for "detach from missile" button.

> The missile should explode after some time after firing and should be able to be shot down by other players

Yea i think there are too many problems with this, just like NOTGreg outlined. Plus yet again, bulletspam weapons gain superiority as they can more effectively defend against missiles. It's also a weird idea in general, why can the homing missile be destroyed but the non-homing one cannot?

> Oh and funny idea the missile's speed should be adjustable.

How would that work exactly?

**Comment by Gregory-Y** [comment]
I suppose you're right in that it would make sense for the missile to only remain active within the player's line of sight, and explode a bit beyond as they lose control of it. I initially wanted it to be a weapon that you could use to kill enemies from across the map but upon further consideration I realized that that makes for bad gameplay.

I also agree with your critique that the missile should not be interceptable by bullets; I'm honestly not sure why I suggested that in the first place.

> How would that work exactly?

You could increase the speed by pressing space, that way the missile goes to space by flying out of the map. yeah actually its a bad idea because that would make balancing difficult.

TLDR: I am "very creative" (very stupid).

**Comment by deferred-impact** [comment]
I mean I'm not against the idea of adjustable missile speed per se, it's just that the game already kinda has too many keybinds, especially as far as mobile players are concerned
